src/shared/conf-parser.c

```diff
@@ -221,8 +221,17 @@ static int parse_line(
                         return -ENOMEM;
 
                 if (sections && !nulstr_contains(sections, n)) {
+                        bool ignore = flags & CONFIG_PARSE_RELAXED;
 
-                        if (!(flags & CONFIG_PARSE_RELAXED) && !startswith(n, "X-"))
+                        ignore = ignore || startswith(n, "X-");
+
+                        if (!ignore) {
+                                _cleanup_free_ char *with_minus = NULL;
+                                with_minus = strjoin("-", n);
+                                ignore = ignore || nulstr_contains(sections, with_minus);
+                        }
+
+                        if (!ignore)
                                 log_syntax(unit, LOG_WARNING, filename, line, 0, "Unknown section '%s'. Ignoring.", n);
```

Comment:
this will break install.c, no? since it doesn't list [Service] and so on?

Reply:
I'm not sure what "this" means here. install.c uses CONFIG_PARSE_RELAXED, and this patch does not change that. A later patch removes CONFIG_PARSE_RELAXED from install.c, but it adds a list of sections at the same time.